httpx/_models.py

```diff
@@ -401,6 +401,72 @@ class Headers(typing.MutableMapping[str, str]):
     HTTP headers, as a case-insensitive multi-dict.
     """
 
+    class _HeadersKeysView(typing.KeysView[str]):
+        def __init__(
+            self, raw: typing.List[typing.Tuple[bytes, bytes]], encoding: str
+        ) -> None:
+            self._raw = raw
+            self._encoding = encoding
+
+        def __contains__(self, key: typing.Any) -> bool:
+            try:
+                normalized_key = key.lower().encode(self._encoding)
+            except AttributeError:
+                return False
+            return any(k == normalized_key for k, _ in self._raw)
+
+        def __iter__(self) -> typing.Iterator[str]:
+            return iter(k.decode(self._encoding) for k, _ in self._raw)
```

Comment:
the generator `k.decode(self._encoding) for k, _ in self.raw` should be fine enough (it is already an `Iterator`)